[trivial but useful] Allow spawner to wait indefinitely if requested

v4hn opened the pull request:
`rospy.wait_for_service` does not allow for `0` to wait indefinitely.
Instead `None` has to be passed explicitly.

A user-provided value of 0 leads to an error message and failure at the moment, so it is alright to use the value to indicate indefinite waiting time.

matthew-reynolds reviewed (changes requested):
Seems reasonable, but since this will make the `timeout` behaviour slightly different than `rospy.wait_for_service`, we should add more info. For example, in the help string for the `timeout` argparse argument, maybe add something along the lines of "If <=0, wait indefinitely".

Alternatively/in addition, we could add a `--no-timeout` argument. Might be clearer to users. Thoughts?

destogl commented:
+1 for --no-timeout

bmagyar commented:
I see value in both using the number to encode wait time (if passed in through a parameter) and the more verbose option. Can we have both please? ;)

v4hn committed:
enable spawner to wait indefinitely if requested

`rospy.wait_for_service` does not allow for `0` to wait indefinitely.
Instead, `None` has to be passed explicitly.

By personal request of three maintainers extended to include documentation
and the alternative argument --no-timeout.

v4hn commented:
Wow, it *is* a trivial patch obviously improving applicability in a case that was not documented *at all* before.
I barely care whether it gets merged or not, I just took a minute to pull-request what I typed locally.

So I'm a bit surprised to get further request for icing and sprinkles. *sigh*
Anyway addressed.

matthew-reynolds reviewed (approved):
Looks great, thanks so much for iterating.

> It is a trivial patch obviously improving applicability in a case that was not documented at all before

Indeed, thanks for pointing out the flaw. Once it was identified, I figured we might as well handle it fully. Always good to strive for the best code we can :)

bmagyar commented:
@v4hn indeed sometimes it is unusual to be on the receiving end of reviews ;)

bmagyar reviewed (approved):
Thanks very much for pointing this out and providing a fix!
